src/main/java/com/griddynamics/cloud/learning/service/ArticleService.java

```diff
@@ -2,15 +2,58 @@
 
 import com.griddynamics.cloud.learning.dao.domain.Article;
 import com.griddynamics.cloud.learning.dao.domain.PaginatedResult;
+import com.griddynamics.cloud.learning.dao.repository.ArticleRepository;
 import com.griddynamics.cloud.learning.web.dto.ArticleFilterRequest;
-
-import java.util.Collections;
+import com.griddynamics.cloud.learning.web.dto.PaginatedRequestParams;
+import com.griddynamics.cloud.learning.web.dto.PurchaseType;
 
 public class ArticleService {
 
-    //TODO (Just a stub for now; part of service implementation task)
+    private final ArticleRepository repository = new ArticleRepository();
+
     public PaginatedResult<Article> getArticles(ArticleFilterRequest requestFilter) {
 
-        return new PaginatedResult<>(Collections.emptyList(), 0);
+        PaginatedResult<Article> articles;
+
+        final PurchaseType purchaseType = requestFilter.getPurchaseType();
+        final String tag = requestFilter.getTag();
+        final String caption = requestFilter.getCaption();
+        final PaginatedRequestParams params = requestFilter.getParams();
+
+        if (!purchaseType.equals(PurchaseType.ALL)) {
+            final Boolean isFree = purchaseType.equals(PurchaseType.FREE) ? Boolean.TRUE : Boolean.FALSE;
```

Comment:
It doesn't make sense to compare enums using equals method, even more, If you look at the source code for .equals(), it simply defers to ==.

P.S. How do you think, why?

Reply:
could be simplified to `Boolean isFree = purchaseType == PurchaseType.FREE`
But we even don't need a dedicated var, let's inline it like this:
`articles = getArticlesByPurchaseType(tag, caption, params, purchaseType == PurchaseType.FREE);`